### PR TITLE
added request response logging middleware

### PR DESCRIPTION
**Pull request by xnetsrak**
## Description

Added nuget package for middleware request response logging.
Added TF setup.
Added implementation registration. 

## References

* #0000

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/Fixtures/ChargesFunctionAppFixture.cs [hunk 1]
```diff
@@ -274,6 +277,15 @@ private async Task InitializeMessageHubAsync()
...
 
+        private void SetUpRequestResponseLogging()
+        {
+            Environment.SetEnvironmentVariable(EnvironmentSettingNames.RequestResponseLoggingConnectionString, ChargesServiceBusResourceNames.RequestResponseLoggingConnectionString);
+            Environment.SetEnvironmentVariable(EnvironmentSettingNames.RequestResponseLoggingContainerName, ChargesServiceBusResourceNames.RequestResponseLoggingContainerName);
+
+            var storage = new BlobContainerClient(ChargesServiceBusResourceNames.RequestResponseLoggingConnectionString, ChargesServiceBusResourceNames.RequestResponseLoggingContainerName);
+            storage.CreateIfNotExists();
+        }
+
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/Fixtures/ChargesFunctionAppFixture.cs:288]: We should make this method async and use `storage.CreateIfNotExistsAsync();`

File: build/infrastructure/main/func-functionhost.tf [hunk 2]
```diff
@@ -71,6 +71,10 @@ module "func_functionhost" {
     MESSAGEHUB_DATAAVAILABLE_QUEUE                                  = "dataavailable"
     MESSAGEHUB_BUNDLEREQUEST_QUEUE                                  = "charges"
     MESSAGEHUB_BUNDLEREPLY_QUEUE                                    = "charges-reply"
+
+    # Shared resources logging
+    REQUEST_RESPONSE_LOGGING_CONNECTION_STRING                      = data.azurerm_key_vault_secret.st_market_operator_logs_primary_connection_string.value
+    REQUEST_RESPONSE_LOGGING_CONATINAER_NAME                        = data.azurerm_key_vault_secret.st_market_operator_logs_container_name.value
```
Comment (BjarkeMeier) [inline build/infrastructure/main/func-functionhost.tf:77]:
```suggestion
    REQUEST_RESPONSE_LOGGING_CONTAINER_NAME                        = data.azurerm_key_vault_secret.st_market_operator_logs_container_name.value
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs [hunk 3]
```diff
@@ -83,6 +84,14 @@ internal static void ConfigureServices(IServiceCollection serviceCollection)
 
             var serviceBusClient = new ServiceBusClient(serviceBusConnectionString);
 
+            serviceCollection.AddScoped<IRequestResponseLogging>(_ =>
+            {
+                var requestResponseLogStorage = EnvironmentHelper.GetEnv(EnvironmentSettingNames.RequestResponseLoggingConnectionString);
+                var requestResponseLogContainerName = EnvironmentHelper.GetEnv(EnvironmentSettingNames.RequestResponseLoggingContainerName);
+
+                return new RequestResponseLoggingBlobStorage(requestResponseLogStorage, requestResponseLogContainerName);
+            });
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs:93]: Would probably be nice to move into a method (like the other stuff).

**Review by BjarkeMeier — APPROVED** [review]
LGTM, but what about tests?

**Pull request merged**